src/azure-cli/azure/cli/command_modules/profile/_subscription_selector.py

```diff
@@ -0,0 +1,108 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core._profile import (_SUBSCRIPTION_NAME, _SUBSCRIPTION_ID, _TENANT_DISPLAY_NAME, _TENANT_ID,
+                                     _IS_DEFAULT_SUBSCRIPTION)
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class SubscriptionSelector:
```

Comment:
Some refactors are made in commit https://github.com/Azure/azure-cli/pull/28910/commits/5884ecb533b877d9ca3c8ac58d72bd4b7bc5685f: 

1. Part of `Profile._interactively_select_subscription` is split into `_format_subscription_table` so that it can be unit-tested.
2. `Profile._interactively_select_subscription` and `_format_subscription_table` are extracted into a `SubscriptionSelector` class.
3. `SubscriptionSelector` is moved to `profile` module to reduce the change on core.